### PR TITLE
Refactor/register identity

### PR DESCRIPTION
**Pull request by erdoganishe**
New Achitecture

### CODE REVIEW

File: circuits/identityManagement/circuits/identity.circom [hunk 1]
```diff
@@ -46,29 +46,36 @@ template RegisterIdentity(
...
+        component isAACheckRSA = IsZero();
+        isAACheckRSA.in <== IS_AA;
+
+        dg15PubKeyHash <== dg15Hasher.out * (1 - isAACheckRSA.out);
 
     } else { // Elliptic Curve Active Auth key extraction
         component xToNum = Bits2Num(248);
         component yToNum = Bits2Num(248);
         
         var EC_FIELD_SIZE = 256;
```
Comment (1KitCat1) [inline circuits/identityManagement/circuits/identity.circom:58]: Check if there are any non-256 AA

File: circuits/identityManagement/circuits/identity.circom [hunk 2]
```diff
@@ -7,67 +7,69 @@ template RegisterIdentity(
...
-
+    
     var DG1_LEN = 1024;
 
-
     signal input dg1[DG1_LEN];                  // 744 || 760 bits + padding
     signal input dg15[DG15_SIZE * HASH_BLOCK_SIZE];                // 1320 || 2096 || 1832 || 2384 || 2520 bits + padding
     signal input skIdentity;
+    if (IS_AA == 1) {
+        if (SIGNATURE_TYPE <= 19) { // rsa keys stored
```
Comment (1KitCat1) [inline circuits/identityManagement/circuits/identity.circom:25]: SIGNATURE_TYPE_AA

Comment (1KitCat1) [inline circuits/identityManagement/circuits/identity.circom:25]: < 20 (same format)

File: circuits/ecdsa/brainpoolP320r1/brainpool.circom [hunk 3]
```diff
@@ -196,7 +196,7 @@ template BrainpoolGetGenerator(CHUNK_SIZE, CHUNK_NUMBER){
     
 }
 
-template GetBrainpoolOrder(CHUNK_SIZE, CHUNK_NUMBER){
+template GetBrainpool320320Order(CHUNK_SIZE, CHUNK_NUMBER){
```
[the remark this answers is not in the document]
Comment (erdoganishe) [reply circuits/ecdsa/brainpoolP320r1/brainpool.circom:199]: done

File: circuits/identityManagement/circuits/identity.circom (hunk 2 above)
Comment (erdoganishe) [reply circuits/identityManagement/circuits/identity.circom:25]: done

Comment (erdoganishe) [reply circuits/identityManagement/circuits/identity.circom:25]: done

File: circuits/passportVerification/passportVerificationBuilder.circom [hunk 4]
```diff
@@ -68,94 +92,66 @@ template PassportVerificationBuilder(
...
-            for (var j = 0; j < HASH_TYPE; j++){
-                encapsulatedContentHash[i-1][j] <== 0;
-            }
+        for (var j = 0; j < DG_HASH_TYPE; j++){
+            dg15Hash[j] <== 0;
         }
     }
 
+    ecPassportHasher = PassportHash(HASH_BLOCK_SIZE, EC_BLOCK_NUMBER, HASH_TYPE);
+    for (var j = 0; j < HASH_BLOCK_SIZE*EC_BLOCK_NUMBER; j++){
```
Comment (1KitCat1) [inline circuits/passportVerification/passportVerificationBuilder.circom:119]: spaces

**Pull request merged**